corehq/apps/app_manager/suite_xml/sections/details.py

```diff
@@ -194,7 +194,11 @@ def build_detail(self, module, detail_type, detail, detail_column_infos, tabs=No
             if len(d.details):
                 helper = EntriesHelper(self.app)
                 datums = helper.get_datum_meta_module(module)
-                d.variables.extend([DetailVariable(name=datum.id, function=datum.datum.value) for datum in datums])
+                d.variables.extend([
+                    DetailVariable(name=datum.id, function=datum.datum.value)
+                    for datum in datums
+                    if datum.action != 'fixture_select'  # FixtureSelect isn't supported under variables
+                ])
```

Comment:
I don't have full insight into how the variables are used by the commcare app, though from the error that was seen in the app which happened when there was a tab in case detail, it appeared that all variables are fetched from the case, and since "@id" was not available on the case, there was an error.
So, it made sense to simply skip fixture select datum while adding variables since it should be needed there but it should be there in the datums so its included in the entry node for the module in suite xml.
This change is limited to the feature flag and should not impact anything outside it.
And even for domains that have this feature flag enabled, should not get impacted since it's a bugfix.

Reply:
I can imagine that if I came across this comment in the code, I would wish that I could read this explanation. Could you extend that comment, and add a link to this thread, or to a Jira ticket, so that the next dev has more context?

Maybe someone with more insight of how variables are used in the app can update this to filter by supported values (maybe?) instead of unsupported values. (On digging into this, I found that `datum.action` is usually a string, [but sometimes an object](https://github.com/dimagi/commcare-hq/blob/fd2f4c89089149b784144fa87fcdef66b5a15212/corehq/apps/app_manager/suite_xml/sections/entries.py#L1131) ... but surely that's for a different type of datum. Naming is hard.)

Reply:
> I would wish that I could read this explanation. Could you extend that comment, and add a link to this thread

Link to this thread seems okay to me. I guess usually I find out where the code is added through git blame but that gets tricky if code is moved around.